### PR TITLE
TT-4964 - Remove GOOS/GOARCH from 3.2

### PR DESCRIPTION
**Pull request by buger**
This changes should end up in release-3

Issue happened because we synced automation from the master

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3965/merge
Commit:  
Triggered by: pull_request (@buger)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2057660701)

File: ci/images/plugin-compiler/data/build.sh [hunk 1]
```diff
@@ -20,22 +20,11 @@ To build a plugin:
...
 fi
 
-# if arch and os present then update the name of file with those params
-if [[ $GOOS != "" ]] && [[ $GOARCH != "" ]]; then
-  plugin_name="${plugin_name%.*}_${CURRENTVERS}_${GOOS}_${GOARCH}.so"
-fi
-
 if [[ $GOOS != "linux" ]];then
     CGOENABLED=1
 fi
```
Comment (sredxny) [inline ci/images/plugin-compiler/data/build.sh:30]: this should be gone as well

**Review by sredxny — CHANGES_REQUESTED** [review]

**Review by sredxny — CHANGES_REQUESTED** [review]
Also the command in the L65 should be replace by:

`GO111MODULE=off go build -buildmode=plugin -o $plugin_name \`

As it should not use anymore `CGO_ENABLE=$CGO_ENABLE GOOS=$GOOS GOARCH=$GOARCH`

**Comment by buger** [comment]
This envs actually do make sense. Removed just file name change. @sredxny pls re-check.

**Review by sredxny — APPROVED** [review]
lgtm

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3965/merge
Commit: 735e90bafcde510ddb8eeb794d071d0961701b67 
Triggered by: pull_request (@buger)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2058214189)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/release-3.2.3
Commit: e938ac144d7b30a4076fe18ecaf2a3d1d23c45c5 TT-4964 - Remove GOOS/GOARCH from 3.2 (#3965)

This changes should end up in release-3

Issue happened because we synced automation from the master
Triggered by: push (@buger)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2058732873)

**Pull request merged**